File: src/main/java/com/gutorov/university/api/news/NewsRs.java
package com.gutorov.university.api.news;

import com.gutorov.university.api.author.AuthorRs;
import com.gutorov.university.api.category.CategoryRs;

import java.time.LocalDateTime;
import java.util.UUID;

public class NewsRs {
    private UUID id;
    private String title;
    private String content;
    private CategoryRs category;
    private AuthorRs author;
    private LocalDateTime postDate;

    public NewsRs() {
    }

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public CategoryRs getCategory() {
        return category;
    }

    public void setCategory(CategoryRs category) {
        this.category = category;
    }

    public AuthorRs getAuthor() {
        return author;
    }

    public void setAuthor(AuthorRs author) {
        this.author = author;
    }

    public LocalDateTime getPostDate() {
        return postDate;
    }

    public void setPostDate(LocalDateTime postDate) {
        this.postDate = postDate;
    }
}

File: src/main/java/com/gutorov/university/api/news/NewsRs.kt
package com.gutorov.university.api.news

import com.gutorov.university.api.author.AuthorRs
import com.gutorov.university.api.category.CategoryRs
import com.gutorov.university.entity.NewsEntity
import java.time.LocalDateTime
import java.util.*
import java.util.stream.StreamSupport

class NewsRs(
    var id: UUID? = null,
    var title: String? = null,
    var content: String? = null,
    var category: CategoryRs? = null,
    var author: AuthorRs? = null,
    var postDate: LocalDateTime? = null
) {
    companion object {
        @JvmStatic
        fun fromEntity(entity: NewsEntity): NewsRs {
            return NewsRs(
                entity.id,
                entity.title,
                entity.content,
                CategoryRs.fromEntity(entity.category),
                AuthorRs.fromEntity(entity.author),
                entity.postDate
            )
        }

        @JvmStatic
        fun fromEntityList(entities: List<NewsEntity>): List<NewsRs> {
            return StreamSupport.stream(entities.spliterator(), false)
                .map(NewsRs::fromEntity)
                .toList();
        }
    }
}
